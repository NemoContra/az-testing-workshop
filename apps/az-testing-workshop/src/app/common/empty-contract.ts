import { Contract } from '@az-testing-workshop/shared/util/api-models';

export const emptyContract: Contract = {
  contractNumber: '',
  id: '',
  start: '',
  end: '',
  premium: 0,
  person: {
    firstname: '',
    lastname: '',
    dateOfBirth: '',
  },
};
