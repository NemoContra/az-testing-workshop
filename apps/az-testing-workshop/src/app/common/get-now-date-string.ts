const fillString = '0' as const;
const getYear = (now: Date) => `${now.getFullYear()}`;
const getMonth = (now: Date) =>
  (now.getMonth() + 1).toString().padStart(2, fillString);
const getDay = (now: Date) => now.getDate().toString().padStart(2, fillString);

export const getNowDateString = () => {
  const now = new Date();
  return [getYear(now), getMonth(now), getDay(now)].join('-');
};
