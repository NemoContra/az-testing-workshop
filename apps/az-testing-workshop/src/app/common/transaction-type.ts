export const transactionTypes = ['AenderungNachname', 'Kuendigung'] as const;
export type TransactionsType = (typeof transactionTypes)[number];

export const transactionQueryParamKey = 'transaktion';
export const transactionQueryParam = <T extends TransactionsType>(
  transactionType: T
) =>
  ({
    [transactionQueryParamKey]: transactionType,
  } as const);
