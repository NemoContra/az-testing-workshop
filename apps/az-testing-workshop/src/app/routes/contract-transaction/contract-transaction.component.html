<nx-link routerLink="/"><nx-icon name="chevron-left" />Zurück</nx-link>
@if(store.errorCode()) {
<nx-error>An error occurred</nx-error>
}
<div class="spinner-outlet">
  @if(store.loading()) {
  <nx-spinner size="medium" />
  }
</div>
@if(store.contract(); as contract) {
<div class="content">
  <contract-display [contract]="contract" />
  <div>
    <nx-formfield label="Transaktion wählen">
      <nx-dropdown
        [value]="store.transactionType()"
        (valueChange)="selectTransactionType($event)"
        placeholder="Bitte wählen"
      >
        @for(transaktionType of transactionTypes; track transaktionType) {
        <nx-dropdown-item [value]="transaktionType">{{
          transaktionType | transactionType
        }}</nx-dropdown-item>
        }
      </nx-dropdown>
    </nx-formfield>
  </div>
  <div>
    @switch (store.transactionType();) { @case ('AenderungNachname') {
    <form (submit)="submitAenderungNachname(nachname.value, contract)">
      <nx-formfield label="Nachname">
        <input
          nxInput
          type="text"
          #nachname
          [value]="contract.person.lastname"
          data-testid="nachname"
        />
      </nx-formfield>
      <button nxButton type="submit">Absenden</button>
    </form>
    } @case ('Kuendigung') {
    <form (submit)="submitKuendigung(contract)">
      <button [disabled]="!!contract.end" nxButton type="submit">
        Kündigen
      </button>
      @if(contract.end) {
      <nx-error class="cancel-error"
        >Der Vertrag wurde bereits zum
        {{ contract.end | date }} gekündigt</nx-error
      >
      }
    </form>
    } }
  </div>
</div>

}
