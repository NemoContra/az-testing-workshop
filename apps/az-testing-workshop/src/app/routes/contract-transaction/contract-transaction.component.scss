nx-link {
  margin: 16px;
  display: flex;
  place-items: center;
}

.spinner-outlet {
  height: 20px;
  display: flex;
  justify-content: flex-end;
}

.content {
  max-width: 1200px;
  margin: auto;
}

.cancel-error {
  margin-top: 16px;
}
