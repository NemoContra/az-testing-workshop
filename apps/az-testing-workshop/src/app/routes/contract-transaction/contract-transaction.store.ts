import { patchState, signalStore, withMethods, withState } from '@ngrx/signals';
import { Contract } from '@az-testing-workshop/shared/util/api-models';
import { rxMethod } from '@ngrx/signals/rxjs-interop';
import { pipe, switchMap, tap } from 'rxjs';
import { tapResponse } from '@ngrx/operators';
import { HttpErrorResponse } from '@angular/common/http';
import { inject } from '@angular/core';
import { TransactionsType } from '../../common/transaction-type';
import { ContractService } from '../../services/contract.service';
import { ContractOverviewStore } from '../contraxt-overview/contract-overview.store';

export type ContractTransactionState = {
  contract: Contract | undefined;
  errorCode: number | undefined;
  loading: boolean;
  transactionType: TransactionsType | undefined;
};

export const initialContractTransactionState: ContractTransactionState = {
  contract: undefined,
  errorCode: undefined,
  loading: false,
  transactionType: undefined,
};

const optimisticUpdateContracts = (
  contracts: Contract[] | undefined,
  updatedContract: Contract
) =>
  contracts?.map((contract) =>
    contract.id === updatedContract.id ? updatedContract : contract
  );

export const ContractTransactionStore = signalStore(
  withState(initialContractTransactionState),
  withMethods(
    (
      store,
      contractService = inject(ContractService),
      { contracts, ...contractOverviewStore } = inject(ContractOverviewStore)
    ) => ({
      selectTransaction: (transactionType: TransactionsType) =>
        patchState(store, { transactionType }),
      getContract: rxMethod<string>(
        pipe(
          tap(() => patchState(store, { loading: true })),
          switchMap((id: string) =>
            contractService.getContract(id).pipe(
              tapResponse({
                next: (contract) => patchState(store, { contract }),
                error: ({ status }: HttpErrorResponse) =>
                  patchState(store, { errorCode: status }),
                finalize: () => patchState(store, { loading: false }),
              })
            )
          )
        )
      ),
      updateContract: rxMethod<Contract>(
        pipe(
          tap(() => patchState(store, { loading: true })),
          switchMap((contract) =>
            contractService.updateContract(contract).pipe(
              tapResponse({
                next: (contract) => {
                  patchState(store, { contract });
                  patchState(contractOverviewStore, {
                    contracts: optimisticUpdateContracts(contracts(), contract),
                  });
                },
                error: ({ status }: HttpErrorResponse) =>
                  patchState(store, { errorCode: status }),
                finalize: () => {
                  patchState(store, { loading: false });
                },
              })
            )
          )
        )
      ),
    })
  )
);
