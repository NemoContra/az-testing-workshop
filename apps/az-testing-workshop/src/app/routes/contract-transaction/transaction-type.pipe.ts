import { Pipe, PipeTransform } from '@angular/core';
import { TransactionsType } from '../../common/transaction-type';

const transactionTypeMap = {
  AenderungNachname: 'Änderung des Nachnamen',
  Kuendigung: 'Vertragskündigung',
} as const satisfies Record<TransactionsType, string>;

export const transformTransactionType = <T extends TransactionsType>(
  transactionType: T
) => transactionTypeMap[transactionType];

@Pipe({
  name: 'transactionType',
  standalone: true,
})
export class TransactionTypePipe implements PipeTransform {
  transform = transformTransactionType;
}
