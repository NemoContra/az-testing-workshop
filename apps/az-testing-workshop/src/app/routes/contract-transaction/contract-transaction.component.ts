import {
  ChangeDetectionStrategy,
  Component,
  inject,
  Input,
} from '@angular/core';
import { ContractTransactionStore } from './contract-transaction.store';
import { NxDropdownModule } from '@aposin/ng-aquila/dropdown';
import { NxFormfieldModule } from '@aposin/ng-aquila/formfield';
import { TransactionTypePipe } from './transaction-type.pipe';
import { NxIconModule } from '@aposin/ng-aquila/icon';
import { NxLinkModule } from '@aposin/ng-aquila/link';
import { Router, RouterLink } from '@angular/router';
import { NxInputModule } from '@aposin/ng-aquila/input';
import { Contract } from '@az-testing-workshop/shared/util/api-models';
import { NxSpinnerModule } from '@aposin/ng-aquila/spinner';
import { NxButtonModule } from '@aposin/ng-aquila/button';
import { FormsModule } from '@angular/forms';
import {
  transactionQueryParam,
  TransactionsType,
  transactionTypes,
} from '../../common/transaction-type';
import { DatePipe } from '@angular/common';
import { NxMessageToastService } from '@aposin/ng-aquila/message';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [
    NxDropdownModule,
    NxFormfieldModule,
    TransactionTypePipe,
    NxIconModule,
    NxLinkModule,
    RouterLink,
    NxInputModule,
    NxSpinnerModule,
    NxButtonModule,
    FormsModule,
    DatePipe,
  ],
  providers: [ContractTransactionStore],
  selector: 'contract-transaction',
  standalone: true,
  styleUrl: './contract-transaction.component.scss',
  templateUrl: './contract-transaction.component.html',
})
export default class ContractTransactionComponent {
  @Input() set id(id: string) {
    this.store.getContract(id);
  }

  @Input() set transaktion(transaktion: TransactionsType | undefined) {
    if (transaktion) {
      this.store.selectTransaction(transaktion);
    }
  }

  store = inject(ContractTransactionStore);
  readonly transactionTypes = transactionTypes;

  private router = inject(Router);
  private messageToastService = inject(NxMessageToastService);

  async selectTransactionType(type: TransactionsType) {
    await this.router.navigate([], {
      queryParams: transactionQueryParam(type),
    });
  }

  submitAenderungNachname(
    lastname: string,
    { person, ...contract }: Contract
  ): void {
    this.store.updateContract({ ...contract, person: { ...person, lastname } });
    this.messageToastService.open(
      'Änderung des Nachname erfolgreich durchgeführt',
      {
        context: 'success',
        duration: 5000,
      }
    );
  }

  submitKuendigung(contract: Contract): void {
    this.store.updateContract({ ...contract, end: new Date().toDateString() });
    this.messageToastService.open('Kündigung erfolgreich durchgeführt', {
      context: 'success',
      duration: 5000,
    });
  }
}
