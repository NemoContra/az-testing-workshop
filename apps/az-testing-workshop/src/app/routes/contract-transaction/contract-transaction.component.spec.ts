import { createRoutingFactory } from '@ngneat/spectator/jest';
import ContractTransactionComponent from './contract-transaction.component';
import { ContractTransactionStore } from './contract-transaction.store';
import { NxMessageToastService } from '@aposin/ng-aquila/message';
import { NxIconModule } from '@aposin/ng-aquila/icon';
import { MockComponent, MockModule } from 'ng-mocks';
import { signal } from '@angular/core';
import { Contract } from '@az-testing-workshop/shared/util/api-models';
import { TransactionsType } from '../../common/transaction-type';
import { mockContracts } from '@az-testing-workshop/shared/util/mock-data';
import { ContractDisplayComponent } from '../../shared/contract-display/contract-display.component';
import { NxDropdownHarness } from '@az-testing-workshop/shared/util/test-harnesses';
import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
import { Router } from '@angular/router';
import { NxErrorModule } from '@aposin/ng-aquila/base';
import { NxFormfieldModule } from '@aposin/ng-aquila/formfield';
import { mockComponent } from '@az-testing-workshop/shared/util/test-helpers';

jest.mock('@aposin/ng-aquila/message', () => ({
  ...jest.requireActual('@aposin/ng-aquila/message'),
  NxMessageToastService: class _NxMessageToastService {
    open = jest.fn(() => ({
      afterDismissed: jest.fn(() => jest.requireActual('rxjs').of(null)),
    }));
  },
}));

jest.mock('../../common/get-now-date-string', () => ({
  getNowDateString: jest.fn(() => '2024-01-01'),
}));

const createState = () => ({
  getContract: jest.fn(),
  selectTransaction: jest.fn(),
  updateContract: jest.fn(),
  errorCode: signal<number | undefined>(undefined),
  loading: signal(false),
  contract: signal<Contract | undefined>(undefined),
  transactionType: signal<TransactionsType | undefined>(undefined),
});

describe('ContractTransactionComponent', () => {
  let store: ReturnType<typeof createState>;

  const createComponent = createRoutingFactory({
    component: ContractTransactionComponent,
    routes: [
      {
        path: ':id',
        component: ContractTransactionComponent,
      },
    ],
    providers: [
      NxMessageToastService,
      {
        provide: ContractTransactionStore,
        useFactory: () => (store = createState()),
      },
    ],
    overrideModules: [
      [NxFormfieldModule, { remove: { exports: [NxErrorModule] } }],
    ],
    overrideComponents: [
      [
        ContractTransactionComponent,
        {
          remove: {
            imports: [NxIconModule, ContractDisplayComponent],
            providers: [ContractTransactionStore],
          },
          add: {
            imports: [
              MockModule(NxIconModule),
              MockComponent(ContractDisplayComponent),
              mockComponent({ selector: 'nx-error' }),
            ],
          },
        },
      ],
    ],
    stubsEnabled: false,
  });

  it('should render in loading store', () => {
    const spectator = createComponent();
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should render with error', () => {
    const spectator = createComponent();
    store.loading.set(false);
    store.errorCode.set(500);
    spectator.detectChanges();
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should render with a contract without transactionType', () => {
    const spectator = createComponent({ props: { id: '123456789' } });

    store.loading.set(false);
    store.contract.set(mockContracts[0]);
    spectator.detectChanges();

    expect(spectator.fixture).toMatchSnapshot();

    expect(store.getContract).toHaveBeenCalledTimes(1);
    expect(store.getContract).toHaveBeenCalledWith('123456789');
    expect(spectator.query(ContractDisplayComponent)?.contract).toEqual(
      mockContracts[0]
    );
  });

  it('should render with a contract and with transactionType "Vertragskündigung"', async () => {
    const spectator = createComponent({
      queryParams: { transaktion: 'Kuendigung' },
      props: { transaktion: 'Kuendigung' },
    });

    store.loading.set(false);
    store.contract.set(mockContracts[0]);
    spectator.detectChanges();

    expect(spectator.fixture).toMatchSnapshot();

    expect(store.selectTransaction).toHaveBeenCalledTimes(1);
    expect(spectator.query(ContractDisplayComponent)?.contract).toEqual(
      mockContracts[0]
    );
  });

  it('should select transactionType "Änderung des Nachnamen"', async () => {
    const spectator = createComponent({ props: { id: '123456789' } });
    const router = spectator.inject(Router);
    jest.spyOn(router, 'navigate');

    store.loading.set(false);
    store.contract.set(mockContracts[0]);
    spectator.detectChanges();

    expect(spectator.query(ContractDisplayComponent)?.contract).toEqual(
      mockContracts[0]
    );

    const loader = TestbedHarnessEnvironment.loader(spectator.fixture);
    const dropdownHarness = await loader.getHarness(NxDropdownHarness);

    expect(await dropdownHarness.getText()).toEqual('Bitte wählen');
    await dropdownHarness.clickItem({ text: 'Änderung des Nachnamen' });

    expect(router.navigate).toHaveBeenCalledTimes(1);
    expect(router.navigate).toHaveBeenCalledWith([], {
      queryParams: {
        transaktion: 'AenderungNachname' satisfies TransactionsType,
      },
    });
  });

  it('should select transactionType "Kündigen"', async () => {
    const spectator = createComponent();
    const router = spectator.inject(Router);
    jest.spyOn(router, 'navigate');

    store.loading.set(false);
    store.contract.set(mockContracts[0]);
    spectator.detectChanges();

    expect(spectator.query(ContractDisplayComponent)?.contract).toEqual(
      mockContracts[0]
    );

    const loader = TestbedHarnessEnvironment.loader(spectator.fixture);
    const dropdownHarness = await loader.getHarness(NxDropdownHarness);

    expect(await dropdownHarness.getText()).toEqual('Bitte wählen');
    await dropdownHarness.clickItem({ text: 'Vertragskündigung' });

    expect(router.navigate).toHaveBeenCalledTimes(1);
    expect(router.navigate).toHaveBeenCalledWith([], {
      queryParams: {
        transaktion: 'Kuendigung' satisfies TransactionsType,
      },
    });
  });

  it('should submit transactionType "Änderung des Nachnamen"', async () => {
    const spectator = createComponent();
    const nxMessageToastService = spectator.inject(NxMessageToastService);

    store.loading.set(false);
    store.contract.set(mockContracts[0]);
    store.transactionType.set('AenderungNachname');
    spectator.detectChanges();

    spectator.typeInElement('Heisterkamp', 'input[data-testid="nachname"]');
    spectator.click('button[type="submit"]');

    expect(store.updateContract).toHaveBeenCalledTimes(1);
    expect(store.updateContract).toHaveBeenCalledWith({
      contractNumber: '1/2345678/9',
      id: '123456789',
      person: {
        dateOfBirth: '1961-05-16',
        firstname: 'Homer',
        lastname: 'Heisterkamp',
      },
      premium: 42.42,
      start: '2024-01-01',
    });

    expect(nxMessageToastService.open).toHaveBeenCalledTimes(1);
    expect(nxMessageToastService.open).toHaveBeenCalledWith(
      'Änderung des Nachname erfolgreich durchgeführt',
      {
        context: 'success',
        duration: 5000,
      }
    );
  });

  it('should submit transactionType "Vertragskündigung"', async () => {
    const spectator = createComponent({
      props: {
        id: '123456789',
      },
    });
    const nxMessageToastService = spectator.inject(NxMessageToastService);

    store.loading.set(false);
    store.contract.set(mockContracts[0]);
    store.transactionType.set('Kuendigung');
    spectator.detectChanges();

    spectator.click('button[type="submit"]');

    expect(store.updateContract).toHaveBeenCalledTimes(1);
    expect(store.updateContract).toHaveBeenCalledWith({
      contractNumber: '1/2345678/9',
      id: '123456789',
      person: {
        dateOfBirth: '1961-05-16',
        firstname: 'Homer',
        lastname: 'Simpson',
      },
      premium: 42.42,
      start: '2024-01-01',
      end: '2024-01-01',
    });

    expect(nxMessageToastService.open).toHaveBeenCalledTimes(1);
    expect(nxMessageToastService.open).toHaveBeenCalledWith(
      'Kündigung erfolgreich durchgeführt',
      {
        context: 'success',
        duration: 5000,
      }
    );
  });

  it('should show an error message if the contract is cancelled', () => {
    const spectator = createComponent();

    store.loading.set(false);
    store.contract.set({ ...mockContracts[0], end: '2024-01-01' });
    store.transactionType.set('Kuendigung');
    spectator.detectChanges();

    expect(spectator.query('nx-error.cancel-error')).toMatchInlineSnapshot(`
      <nx-error
        class="cancel-error"
      >
        Der Vertrag wurde bereits zum Jan 1, 2024 gekündigt
      </nx-error>
    `);
  });
});
