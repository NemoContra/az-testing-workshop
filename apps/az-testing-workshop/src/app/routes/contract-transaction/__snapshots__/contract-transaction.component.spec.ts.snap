// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractTransactionComponent should render in loading store 1`] = `
<contract-transaction
  messageToastService={[Function _NxMessageToastService]}
  router={[Function _Router]}
  store={[Function Object]}
  transactionTypes={[Function Array]}
>
  <nx-link
    class="nx-link nx-link--small"
    routerlink="/"
    tabindex="0"
  >
    <nx-icon
      class="nx-link__icon"
      name="chevron-left"
    />
    Zurück
  </nx-link><div
    class="spinner-outlet"
  >
  </div>
</contract-transaction>
`;

exports[`ContractTransactionComponent should render with a contract and with transactionType "Vertragskündigung" 1`] = `
<contract-transaction
  messageToastService={[Function _NxMessageToastService]}
  router={[Function _Router]}
  store={[Function Object]}
  transactionTypes={[Function Array]}
>
  <nx-link
    class="nx-link nx-link--small"
    routerlink="/"
    tabindex="0"
  >
    <nx-icon
      class="nx-link__icon"
      name="chevron-left"
    />
    Zurück
  </nx-link><div
    class="spinner-outlet"
  >
  </div><div
    class="content"
  >
    <contract-display />
    <div>
      <nx-formfield
        class="nx-formfield--type-nx-dropdown nx-formfield is-floating"
        label="Transaktion wählen"
      >
        <div
          class="nx-formfield__wrapper"
        >
          <div
            class="nx-formfield__row"
          >
            <div
              class="nx-formfield__flexfield"
            >
              <div
                class="nx-formfield__input-container"
              >
                <div
                  class="nx-formfield__input"
                >
                  <span
                    class="nx-formfield__label-holder"
                  >
                    <label
                      aria-owns="nx-dropdown-rendered-2"
                      class="nx-formfield__label"
                      for="nx-dropdown-rendered-2"
                      id="nx-formfield-label-1"
                    >
                      <span>
                        Transaktion wählen
                      </span>
                    </label>
                  </span>
                  <nx-dropdown
                    aria-controls="nx-dropdown-modal-3"
                    aria-disabled="false"
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-invalid="false"
                    aria-labelledby="nx-formfield-label-1"
                    aria-required="false"
                    class="nx-dropdown"
                    placeholder="Bitte wählen"
                    role="combobox"
                    tabindex="0"
                  >
                    <div
                      cdkoverlayorigin=""
                      class="nx-dropdown__container"
                    >
                      <div
                        class="nx-dropdown__rendered"
                        id="nx-dropdown-rendered-2"
                      >
                        <span>
                          Bitte wählen
                        </span>
                      </div>
                      <div
                        class="nx-dropdown__icon"
                      >
                        <nx-icon
                          class="nx-dropdown__chevron nx-icon--auto"
                          data-nx-icon-name="chevron-down"
                          name="chevron-down"
                        >
                          <svg
                            fill="none"
                            height="1em"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M6.34317 7.75732L4.92896 9.17154L12 16.2426L19.0711 9.17157L17.6569 7.75735L12 13.4142L6.34317 7.75732Z"
                              fill="currentColor"
                            />
                          </svg>
                        </nx-icon>
                      </div>
                    </div>
                  </nx-dropdown>
                </div>
              </div>
            </div>
          </div>
        </div>
      </nx-formfield>
    </div>
    <div>
    </div>
  </div>
</contract-transaction>
`;

exports[`ContractTransactionComponent should render with a contract without transactionType 1`] = `
<contract-transaction
  messageToastService={[Function _NxMessageToastService]}
  router={[Function _Router]}
  store={[Function Object]}
  transactionTypes={[Function Array]}
>
  <nx-link
    class="nx-link nx-link--small"
    routerlink="/"
    tabindex="0"
  >
    <nx-icon
      class="nx-link__icon"
      name="chevron-left"
    />
    Zurück
  </nx-link><div
    class="spinner-outlet"
  >
  </div><div
    class="content"
  >
    <contract-display />
    <div>
      <nx-formfield
        class="nx-formfield--type-nx-dropdown nx-formfield is-floating"
        label="Transaktion wählen"
      >
        <div
          class="nx-formfield__wrapper"
        >
          <div
            class="nx-formfield__row"
          >
            <div
              class="nx-formfield__flexfield"
            >
              <div
                class="nx-formfield__input-container"
              >
                <div
                  class="nx-formfield__input"
                >
                  <span
                    class="nx-formfield__label-holder"
                  >
                    <label
                      aria-owns="nx-dropdown-rendered-0"
                      class="nx-formfield__label"
                      for="nx-dropdown-rendered-0"
                      id="nx-formfield-label-0"
                    >
                      <span>
                        Transaktion wählen
                      </span>
                    </label>
                  </span>
                  <nx-dropdown
                    aria-controls="nx-dropdown-modal-1"
                    aria-disabled="false"
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-invalid="false"
                    aria-labelledby="nx-formfield-label-0"
                    aria-required="false"
                    class="nx-dropdown"
                    placeholder="Bitte wählen"
                    role="combobox"
                    tabindex="0"
                  >
                    <div
                      cdkoverlayorigin=""
                      class="nx-dropdown__container"
                    >
                      <div
                        class="nx-dropdown__rendered"
                        id="nx-dropdown-rendered-0"
                      >
                        <span>
                          Bitte wählen
                        </span>
                      </div>
                      <div
                        class="nx-dropdown__icon"
                      >
                        <nx-icon
                          class="nx-dropdown__chevron nx-icon--auto"
                          data-nx-icon-name="chevron-down"
                          name="chevron-down"
                        >
                          <svg
                            fill="none"
                            height="1em"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M6.34317 7.75732L4.92896 9.17154L12 16.2426L19.0711 9.17157L17.6569 7.75735L12 13.4142L6.34317 7.75732Z"
                              fill="currentColor"
                            />
                          </svg>
                        </nx-icon>
                      </div>
                    </div>
                  </nx-dropdown>
                </div>
              </div>
            </div>
          </div>
        </div>
      </nx-formfield>
    </div>
    <div>
    </div>
  </div>
</contract-transaction>
`;

exports[`ContractTransactionComponent should render with error 1`] = `
<contract-transaction
  messageToastService={[Function _NxMessageToastService]}
  router={[Function _Router]}
  store={[Function Object]}
  transactionTypes={[Function Array]}
>
  <nx-link
    class="nx-link nx-link--small"
    routerlink="/"
    tabindex="0"
  >
    <nx-icon
      class="nx-link__icon"
      name="chevron-left"
    />
    Zurück
  </nx-link><nx-error>
    An error occurred
  </nx-error><div
    class="spinner-outlet"
  >
  </div>
</contract-transaction>
`;
